File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/Quit_Request.java
package org.emulinker.kaillera.controller.v086.protocol;

import com.google.auto.value.AutoValue;
import org.emulinker.kaillera.controller.messaging.MessageFormatException;

@AutoValue
public abstract class Quit_Request extends Quit {
  private static final String DESC = "User Quit Request";

  private static final int USER_ID = 0xFFFF;
  private static final String USERNAME = "";

  public static Quit_Request create(int messageNumber, String message)
      throws MessageFormatException {
    V086Message.validateMessageNumber(messageNumber, DESC);

    return new AutoValue_Quit_Request(messageNumber, ID, DESC, USERNAME, USER_ID, message);
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/Quit_Request.kt
package org.emulinker.kaillera.controller.v086.protocol

import org.emulinker.kaillera.controller.messaging.MessageFormatException
import org.emulinker.kaillera.controller.v086.protocol.V086Message.Companion.validateMessageNumber

data class Quit_Request
    @Throws(MessageFormatException::class)
    constructor(override val messageNumber: Int, override val message: String) : Quit() {

  override val username = ""
  override val userId = 0xFFFF

  override val messageId = ID

  init {
    validateMessageNumber(messageNumber)
  }
}
